### PR TITLE
reorganize source code

### PR DESCRIPTION
**Pull request by greenknot**
The current source code architecture is messy.

Move everything related to syscall implementations to `bolos/` and move half of `launcher.c` to `svc.c`.

### CODE REVIEW

File: src/bolos/cx_aes.c [hunk 1]
```diff
@@ -1,13 +1,13 @@
-#include <assert.h>
 #include <err.h>
+#include <assert.h>
```
Comment (niooss-ledger) [inline src/bolos/cx_aes.c:2]: Why did you change the order of the headers? `clang-format` sorts the headers alphabetically, by default. Did you use another option to sort them differently?

File: src/svc.c [hunk 2]
```diff
@@ -0,0 +1,283 @@
+#define _GNU_SOURCE
```
Comment (niooss-ledger) [inline src/svc.c:1]: To better know why this is needed, I suggest adding a comment:
```c
#define _GNU_SOURCE /* for memmem() */
```

File: src/sdk.h [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+typedef enum {
+  SDK_NANO_X_1_2,
+  SDK_NANO_S_1_5,
+  SDK_NANO_S_1_6,
+  SDK_BLUE_2_2_5,
+  SDK_LAST
```
Comment (niooss-ledger) [inline src/sdk.h:8]: `SDK_LAST` is misleading, as this value is "invalid", or the "count"/"numbers" of versions.

This PR did not introduce `SDK_LAST` (it was previously in `src/emulate.h`), but this PR could be the opportunity to rename this value to something such as `SDK_COUNT` (to indicate that this value is both the last one of `sdk_version_t` so it is suitable in a `for` loop, and that this value is not valid, because it is "one above the last valid version").

File: src/svc.h [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include <signal.h>
```
Comment (niooss-ledger) [inline src/svc.h:3]: This file used types that require more `#include`: `bool`, `size_t`... Please add:
```c
#include <stdbool.h>
#include <stddef.h>
```

**Review by niooss-ledger — COMMENTED** [review]
This looks good :+1:. I commented on some minor things that should be changed before merging, IMHO.

File: src/svc.h (hunk 4 above)
Comment (niooss-ledger) [reply src/svc.h:3]: ... I have directly done this.

File: src/bolos/cx_aes.c (hunk 1 above)
Comment (greenknot) [reply src/bolos/cx_aes.c]: No reason...

File: src/sdk.h (hunk 3 above)
Comment (greenknot) [reply src/sdk.h]: Done

**Comment by greenknot** [comment]
Thanks @niooss-ledger! I force-pushed the suggested changes.

**Comment by greenknot** [comment]
Rebased and merged manually: 9862118edcc3deb0fa8149e210edc87336a2b008.
